addons/event_sale/models/event_event.py

```diff
@@ -28,22 +28,21 @@ def _compute_sale_price_subtotal(self):
         have to do one conversion per sale.order (and a sale.order is created every time
         we sell a single event ticket). """
         date_now = fields.Datetime.now()
-        event_subtotals = self.env['sale.order.line']._read_group(
-            [('event_id', 'in', self.ids), ('price_subtotal', '!=', 0), ('state', '!=', 'cancel')],
-            ['event_id', 'currency_id'],
-            ['price_subtotal:sum'],
+        event_totals = self.env['sale.order.line']._read_group(
+            [('event_id', 'in', self.ids), ('price_total', '!=', 0), ('state', '!=', 'cancel')],
+            ['event_id', 'currency_id', 'order_id'],
         )
-        event_subtotals_mapping = dict.fromkeys(self._origin, 0)
-        for event, currency, sum_price_subtotal in event_subtotals:
-            event_subtotals_mapping[event] += event.currency_id._convert(
-                sum_price_subtotal,
+        event_totals_mapping = dict.fromkeys(self._origin, 0)
+        for event, currency, order_id in event_totals:
+            event_totals_mapping[event] += event.currency_id._convert(
+                order_id.amount_total,
```

Comment:
Hello @thsh-odoo 
Sorry for the delay on this one.

I had another look and I think that there is one big issue with doing this.
This completely ignore the sale order line meaning that the value used could include anything (other event registrations, unrelated products, ...)

If we want to include possible discount order lines then we would need to search them base on the product linked to the discount (don't know if it's easily done).

Can you have a look ?
Cheers!

Reply:
Yes, I thought that too.
will look into this more and let you know :smiley: 
Thanks!!

Reply:
Hello @jeh-odoo 

I tried to do it easily but didn't find anything 
Because when the **global discount or fixed amount** is added the event_id is empty 
So I tried to add the domain `['|', ('event_id', 'in', self.ids), ('price_total', '<', 0)] `
But still, it is not accurate 
Can you please share your thoughts on this :thinking: 

Thanks!! :slightly_smiling_face:

Reply:
Hello @thsh-odoo 

Thank you for having a deep look on this one. I've just checked a bit on my side as well and have the same conclusion. I don't think that there is a clean way to take into account the discount when on his own order line.

We could technically check and get the `sale_discount_product_id` of the company and filter the order line based on the product. But then in the case of a global discount we would need to recompute the true value discounted for the order line linked to a registration (as we could have other order line in the sale order) which will add another complexity for just a stat widget. And in case of a fixed amount, it's hard to know what is linked to the registrations.

IMO, we can ignore these cases, as it's not worth it. We can only take into account the discount applied to the sale order lines directly (either done with the button manually or based on the pricelist). In the end, it makes senses that we can only take into account the price linked to registrations for the widget and trying to take into account the other will only lead to issues in the end.

So here I would done what you had before which was ignoring the sale order and just use the `price_total` field :slightly_smiling_face: 

Cheers!

Reply:
Hello @jeh-odoo 

Changes are reverted.
Can you please review it again?

Thanks!!